### PR TITLE
drop support for h-as-* classes, both incoming and outgoing

### PR DESCRIPTION
**Pull request by snarfed**
...they're deprecated in favor of https://indiewebcamp.com/post-type-discovery

hey @kylewm you're better at mf2 than me...mind giving this a once over? tia!

fixes snarfed/bridgy#635

### CODE REVIEW

File: granary/microformats2.py [hunk 1]
```diff
@@ -268,12 +242,10 @@ def json_to_object(mf2):
...
-  for h_class, mf2_type in h_class_overrides:
-    if h_class in mf2.get('type', []):
-      break  # found
-  else:
-    mf2_type = mf2util.post_type_discovery(mf2)
-
+  mf2_types = mf2.get('type') or []
+  mf2_type = ('location' if 'h-geo' in mf2_types
+              else 'person' if 'h-card' in mf2_types
+              else mf2util.post_type_discovery(mf2))
```
Comment (kylewm) [inline granary/microformats2.py]: `mf2util.post_type_discovery` should return 'person' for h-cards (or 'org' if the h-card has an org property that is the same as its name). I'd be more than happy to add 'h-geo' => location too.

File: granary/test/test_facebook.py [hunk 2]
```diff
@@ -831,7 +831,7 @@ def tag_uri(name):
 </a>
 <span class="summary">my link caption</span>
 </p>
-<span class="p-location h-card h-as-location">
+<span class="p-location h-card h-geo">
```
Comment (kylewm) [inline granary/test/test_facebook.py]: I don't think this is exactly semantically correct per [h-card](http://microformats.org/wiki/h-card). 

It says h-card commonly embeds a "p-geo h-geo" or "p-adr h-adr", but I don't think it is meaningful for one thing to be both an "h-card" and an "h-geo".

**Comment by kylewm** [comment]
woohoo, love it. and it looks good other than my question about the "h-card h-geo" thing; pretty sure we've talked about it in #microformats before ... need a card-type-discovery analogue to the post-type-discovery algorithm.

File: granary/microformats2.py (hunk 1 above)
Comment (kylewm) [inline granary/microformats2.py]: amending that per my other comment: happy to add "person", "location", etc. to the `mf2util.post_type_discovery` algorithm, once we work out what that algorithm is

Comment (snarfed) [inline granary/microformats2.py]: yay thanks! updated.

File: granary/test/test_facebook.py (hunk 2 above)
Comment (snarfed) [inline granary/test/test_facebook.py]: totally true, i'm overusing `h-geo`. i'll walk that back.

**Comment by kylewm** [comment]
@coveralls please, continue to keep us appraised of fluctuations of thousandths of a percent.

**Comment by snarfed** [comment]
oh snap!

**Pull request merged**